### PR TITLE
Add Sinc Rev. 4

### PR DESCRIPTION
**Pull request by nooges**
## Description

Add Sinc Rev. 4

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/sinc/rev4/halconf.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_I2C TRUE
```
Comment (drashna) [inline keyboards/keebio/sinc/rev4/halconf.h:19]: I2C enabled, but not used?

**Review by drashna — APPROVED** [review]

File: keyboards/keebio/sinc/rev4/config.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (zvecr) [inline keyboards/keebio/sinc/rev4/config.h:15]:
```suggestion
// Copyright 2023 Danny Nguyen (danny@keeb.io)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Probably best to avoid mixing header styles

File: keyboards/keebio/sinc/rev4/halconf.h (hunk 1 above)
Comment (zvecr) [reply keyboards/keebio/sinc/rev4/halconf.h:6]: @nooges any update to this?

Comment (nooges) [reply keyboards/keebio/sinc/rev4/halconf.h:6]: I've got breakout pins for I2C, but it's not being actively used, so should I just set to `FALSE` and let someone else override?

Comment (zvecr) [reply keyboards/keebio/sinc/rev4/halconf.h:6]: I see no harm in it being configured, from my perspective it was more about making sure you had not missed anything. Others might feel differently i guess?

**Pull request merged**